beacon-chain/sync/initial-sync/round_robin.go

```diff
@@ -162,6 +170,10 @@ func (s *InitialSync) roundRobinSync(genesis time.Time) error {
 
 		for _, blk := range blocks {
 			logSyncStatus(genesis, blk, peers, counter)
+			if !s.chain.BlockExists(ctx, bytesutil.ToBytes32(blk.ParentRoot)) {
+				log.Debugf("Beacon node doesn't have a block in db with root %#x", blk.ParentRoot)
+				continue
```

Comment:
```suggestion
				break
				blocks:= blocks[:i-1]
```

Reply:
we need it continue, if we get a forked block we just ignore it

Reply:
but we need the slot that is missing

Reply:
It would fall over to the next request yes, but this way the node will not be stalled if we receive a forked block. If you try and request parents for those blocks, a forked block can end up stalling the node, as you request its parents , and its ancestors, etc.

Reply:
sg